addons/account/models/account_move.py

```diff
@@ -3788,18 +3785,10 @@ def default_get(self, default_fields):
 
         if 'account_id' in default_fields \
             and (self._context.get('journal_id') or self._context.get('default_journal_id')) \
-            and not values.get('account_id') \
-            and self._context.get('default_move_type') in self.move_id.get_inbound_types():
+            and not values.get('account_id'):
             # Fill missing 'account_id'.
             journal = self.env['account.journal'].browse(self._context.get('default_journal_id') or self._context['journal_id'])
-            values['account_id'] = journal.default_credit_account_id.id
-        elif 'account_id' in default_fields \
-            and (self._context.get('journal_id') or self._context.get('default_journal_id')) \
-            and not values.get('account_id') \
-            and self._context.get('default_move_type') in self.move_id.get_outbound_types():
-            # Fill missing 'account_id'.
-            journal = self.env['account.journal'].browse(self._context.get('default_journal_id') or self._context['journal_id'])
-            values['account_id'] = journal.default_debit_account_id.id
```

Comment:
Your cleanup only works for invoices but not with miscellaneous operations. You need to restore `and self._context.get('default_move_type') in ('out_invoice', 'out_refund', 'in_invoice', 'in_refund', 'out_receipt', 'in_receipt')`

Reply:
I have 2 remarks here:
- `(self.move_id.get_inbound_types(True) + self.move_id.get_outbound_types(True))` is exactly the same as `self.move_id.is_invoice(include_receipts=True)`
- `default_get` is an `api.model` methods so you can't use `self.move_id`. Note: It works by magic actually.

So, this is the reason why the only way to check you are dealing with an invoice is to use:
```
self._context.get('default_move_type') in ('out_invoice', 'out_refund', 'in_invoice', 'in_refund', 'out_receipt', 'in_receipt')
```